dashboard/src/components/AppUpgrade/AppUpgrade.tsx

```diff
@@ -77,11 +87,33 @@ class AppUpgrade extends React.Component<IAppUpgradeProps, IAppUpgradeState> {
             resource="App Repositories"
           />
         );
+      } else if (repos.length === 0) {
```

Comment:
I couldn't see a test case for when repos are empty, can we add that if it isn't there?